### PR TITLE
content: How Mercado Libre optimized for Web Vitals (TBT, FID)

### LINKED ISSUE #3566: content: MercadoLibre [2020-08-17]
* [draft](https://docs.google.com/document/d/1u6U5sFTzK6j8ny9jN6HKFWrm5CjFxMzKl2FwbeoZjwc/edit)
* [thread](https://groups.google.com/a/google.com/g/webdev-content/c/UZjovFQ0X1U)
* next step: @demianrenzulli is working with partner to see if any technical content can be expanded upon

### PR DESCRIPTION
**Pull request by demianrenzulli**
Fixes #3566 

Case study for Mercado Libre's FID optimization.

cc // @kaycebasques

### CODE REVIEW

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
+---
+layout: post
+title: How Mercado Libre optimized for Web Vitals (TBT/FID)
+subhead: Optimizing interactivity of Mercado Libre's product details pages.
+authors:
+  - demianrenzulli
+  - aranhacarlos
+  - joanbaca
+date: 2020-09-01
+hero: hero.jpg
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:10]: The hero image looks cutoff on web.dev/blog in an awkward way:

image

Do you want to create a thumbnail that focuses on the Mercado Libre logo? https://web.dev/handbook/markup-media/#thumbnail

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+description: Summary of the work done by Mercado Libre's frontend architecture team to optimize FID, using TBT as a proxy metric.
+tags:
+  - blog
+  - performance
+  - web-vitals
+---
+
+Mercado Libre is the largest e-commerce and payments ecosystem in Latin America. It is present in 18 countries and is a market leader in Brazil, Mexico, and Argentina (based on unique visitors and pageviews).
+
+Web performance has been a focus for the company for a long time. They recently formed a team to monitor performance and apply optimizations across different parts of the site.
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:20]:
```suggestion
Web performance has been a focus for the company for a long time, but they recently formed a team to monitor performance and apply optimizations across different parts of the site.
```

I noticed this in the draft but didn't know how to address it. But the answer is obvious now. We need to make the ideas in these two sentences flow into each other more. My suggested rephrasing does that. They created a dedicated performance team, but they were already focused on performance for a long time.

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+This article summarizes the work done by [Guille Paz](https://twitter.com/pazguille), [Pablo Carminatti](https://www.linkedin.com/in/pcarminatti/), and [Oleh Burkhay](https://twitter.com/oburkhay) from Mercado Libre's frontend architecture team to optimize one of the Core Web Vitals: [First Input Delay (FID)](https://web.dev/fid/) and it's lab proxy, [Total Blocking Time (TBT)](https://web.dev/tbt/).
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:22]:
```suggestion
This article summarizes the work done by [Guille Paz](https://twitter.com/pazguille), [Pablo Carminatti](https://www.linkedin.com/in/pcarminatti/), and [Oleh Burkhay](https://twitter.com/oburkhay) from Mercado Libre's frontend architecture team to optimize one of the Core Web Vitals: [First Input Delay (FID)](https://web.dev/fid/) and its lab proxy, [Total Blocking Time (TBT)](https://web.dev/tbt/).
```
it's = it is (which is not what we mean here)

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:23]: Can you add a [Stats component](https://web.dev/handbook/web-dev-components/#stats) here that summarizes the main successes?

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+While Mercado Libre's site was performing well in most sections, they found in the [Chrome User Experience Report](https://developers.google.com/web/tools/chrome-user-experience-report) that product detail pages had a poor FID. Based on that information, they decided to focus their efforts on improving the interactivity for product pages in the site.
+
+<figure class="w-figure">
+  <img src="meli-pdp.png" 
+       alt="The mobile and desktop versions of a Mercado Libre product detail page.">
+   <figcaption class="w-figcaption">
+      The mobile and desktop versions of a Mercado Libre product detail page.
+  </figcaption>     
+</figure>
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:48]: Can you split this up into two images? The [Column compontent](https://web.dev/handbook/web-dev-components/#columns) should help you keep them side-by-side.

And make sure to add `class="w-screenshot"` to each image please

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+These pages allow the user to perform complex interactions, so the goal was optimization interactivity, without removing valuable functionality.
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:50]:
```suggestion
These pages allow the user to perform complex interactions, so the goal was interactivity optimization, without interfering with valuable functionality.
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+{% Aside %}
+During this project Mercado Libre used [Max Potential FID](https://web.dev/lighthouse-max-potential-fid/) in Lighthouse because that was the tool's main metric for measuring interactivity at that time. Lighthouse now recommends using [Total Blocking Time](https://web.dev/tbt/) instead.
+{% endAside %}
+
+## Use WebPageTest to visualize long tasks
+
+WebPageTest (WPT) is a web performance tool that allows you to run tests on real devices in different locations around the world.
+
+Mercado Libre used WPT to reproduce the experience of their users by choosing a device type and location similar to real users. Specifically, they chose a **Moto 4G device** and **Dulles, Virginia**, because they wanted to approximate the experience of Mercado Libre users in Mexico.
+By observing the main thread view of WPT, Mercado Libre found that there were several consecutive long tasks blocking the man thread for 2 seconds:
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:76]:
```suggestion
By observing the main thread view of WPT, Mercado Libre found that there were several consecutive long tasks blocking the main thread for 2 seconds:
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+
+<figure class="w-figure">
+  <img src="main-thread-unoptimized.png"
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:79]:
```suggestion
  <img src="main-thread-unoptimized.png" class="w-screenshot"
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+       alt="Main thread view of Mercado Libre's product detail pages.">
+   <figcaption class="w-figcaption">
+      Main thread view of Mercado Libre's product detail pages.
+  </figcaption>     
+</figure>
+
+Analyzing the corresponding waterfall they found that:
+
+A considerable part of those two seconds came from their analytics module.
+The main bundle size of the application was large (950KB) and took a long time to parse, compile, and execute.
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:89]:
```suggestion
Analyzing the corresponding waterfall they found that
a considerable part of those two seconds came from their analytics module.
The main bundle size of the application was large (950KB) and took a long time to parse, compile, and execute.
```

The colon (`they found that:`) makes it seem like a list will follow

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+<figure class="w-figure">
+  <img src="waterfall-unoptimized.png"
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:92]:
```suggestion
  <img src="waterfall-unoptimized.png" class="w-screenshot"
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+
+- Initially they were requiring the full [Lodash module](https://lodash.com/). This was replaced with a [per-method require](https://lodash.com/per-method-packages) to load only a subset of Lodash instead of the whole library, and used in conjunction with [lodash-webpack-plugin](https://github.com/lodash/lodash-webpack-plugin) to shrink Lodash even further.
+
+They also applied the following [Babel](https://babeljs.io/) optimizations:
+
+- Using [@babel/plugin-transform-runtime](https://babeljs.io/docs/en/babel-plugin-transform-runtime) to reuse Babel's helpers throughout the code, and reduce the size of the bundle considerably.
+- Using [babel-plugin-search-and-replace](https://github.com/jean-smaug/babel-plugin-search-and-replace#readme) to replace tokens at build time, in order to remove a large configuration file inside the main bundle.
+- Adding [babel-plugin-transform-react-remove-prop-types](https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types#readme) to save some extra bytes by removing the prop types.
+
+As a result of these optimizations, the bundle size was reduced **by approximately 16%**.
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:132]: Is this in addition to the 2% reduction that you mentioned earlier (so 18% total)? We should clarify this.

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+
+## Measure impact
+
+The changes lowered Mercado Libre's consecutive long tasks **from two seconds to one second**:
+
+<figure class="w-figure">
+  <img src="main-thread-iteration-1.png"
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:139]:
```suggestion
  <img src="main-thread-iteration-1.png" class="w-screenshot"
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+
+## Measure impact
+
+The changes lowered Mercado Libre's consecutive long tasks **from two seconds to one second**:
+
+<figure class="w-figure">
+  <img src="main-thread-iteration-1.png" 
+       alt="Main thread view of Mercado Libre's product detail pages after first round of optimizations.">
+   <figcaption class="w-figcaption">
+      In the top waterfall of WPT there’s a long red bar (in the Page is Interactive row) between seconds 2 and 5. In the bottom waterfall, the bar has been broken into smaller pieces, occupying the main thread for shorter periods of time.
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:142]:
```suggestion
      In the top waterfall of WPT there’s a long red bar (in the <b>Page is Interactive</b> row) between seconds 2 and 5. In the bottom waterfall, the bar has been broken into smaller pieces, occupying the main thread for shorter periods of time.
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+  </figcaption>     
+</figure>
+
+Lighthouse showed a **50% reduction** in Max Potential First Input Delay:
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:146]:
```suggestion
Lighthouse showed a **57% reduction** in Max Potential First Input Delay:
```
We should use the precise number. 1710ms to 750ms. I think that's 57% but you should double-check my math.

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+  <img src="lighthouse-iteration-1.png" 
+       alt="Lighthouse metrics in a PSI report for Mercado Libre's product detail pages after first round of optimizations.">
+</figure>
+
+## Second iteration
+
+The team continued digging into long tasks in order to find subsequent improvements.
+
+<figure class="w-figure">
+  <img src="main-thread-iteration-1-detail.png"
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:158]:
```suggestion
  <img src="main-thread-iteration-1-detail.png" class="w-screenshot"
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 16]
```diff
@@ -0,0 +1,212 @@
...
+       alt="Detailed view of main thread view of Mercado Libre's product detail pages after first round of optimizations.">
+   <figcaption class="w-figcaption">
+      The Waterfall (not pictured) helped Mercado Libre identify which libraries were using the main thread heavily (Browser Main Thread row) and the Page is Interactive row clearly shows that this main thread activity is blocking interactivity.
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:161]:
```suggestion
      The Waterfall (not pictured) helped Mercado Libre identify which libraries were using the main thread heavily (<b>Browser Main Thread</b> row) and the <b>Page is Interactive</b> row clearly shows that this main thread activity is blocking interactivity.
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+
+Based on that information they decided to implement the following changes:
+
+- Continue reducing the main bundle size to optimize compile and parse time (e.g. by removing duplicate dependencies throughout the different modules).
+- Apply [code splitting](https://web.dev/reduce-javascript-payloads-with-code-splitting/) at component level, to divide JavaScript in smaller chunks and allow for smarter loading of the different components.
+- Defer [component hydration](https://developers.google.com/web/updates/2019/02/rendering-on-the-web#rehydration) to allow for a smarter use of the main thread. This technique is commonly referred to as [partial hydration](https://addyosmani.com/blog/rehydration/).
+
+## Measure impact
+
+The resulting Webpagetest trace showed even smaller chunks of JS execution:
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:173]:
```suggestion
The resulting WebPageTest trace showed even smaller chunks of JS execution:
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+
+<figure class="w-figure">
+  <img src="main-thread-iteration-2.png"
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:176]:
```suggestion
  <img src="main-thread-iteration-2.png" class="w-screenshot"
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 19]
```diff
@@ -0,0 +1,212 @@
...
+       alt="Main thread view of Mercado Libre's product detail pages after secoond round of optimizations.">
+</figure>
+
+And their Max Potential FID time in Lighthouse was reduced **by an additional 60%**:
+
+<figure class="w-figure">
+  <img src="lighthouse-iteration-2.png"
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:183]:
```suggestion
  <img src="lighthouse-iteration-2.png" class="w-screenshot"
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 20]
```diff
@@ -0,0 +1,212 @@
...
+  - aranhacarlos
+  - joanbaca
+date: 2020-09-01
+hero: hero.jpg
+description: Summary of the work done by Mercado Libre's frontend architecture team to optimize FID, using TBT as a proxy metric.
+tags:
+  - blog
+  - performance
+  - web-vitals
+---
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:16]:
```suggestion
  - web-vitals
  - case-study
  - lighthouse
  - chrome-ux-report
  - javascript
---
```

**Review by kaycebasques — CHANGES_REQUESTED** [review]
- [ ] Please break all lines at 100 characters. If you're using VS Code, [Rewrap](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) makes it pretty easy to do. The [Docs To X](https://gsuite.google.com/u/0/marketplace/app/docs_to_x/243920501110?hl=en&pann=docs_addon_widget) Google Docs Add-On also supports this.

Reminder that it's much faster to commit my requested changes by going to **Files changed** tab and using the **Add suggestion to batch** feature (rather than committing suggestions one-by-one).

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 11 above)
Comment (demianrenzulli) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:135]: Hi @kaycebasques! I know this part is confusing... In summary:

- The first optimization obtained a 2% reduction in JS for the whole site for **the internal tracking module**.
- The second optimization, obtained a 16% reduction in JS For **the main bundle loaded by PDP pages**.

I don't think we should consider that these optimizations should be added as 2% + 16% = 18%, as they have been applied to different assets, but, form your comments, the way it's expressed right now is not clear.

Maybe you have a suggestion on how to structure this part better?

Thanks a lot!

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 5 above)
Comment (demianrenzulli) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:52]: Thanks @kaycebasques! I tried the following:

- Breaking the main image into two independent images (one for mobile the other for desktop).
- Using the column component.

In the original version, the mobile screenshot takes around 1/3 of the width. When using the component it assigned half of the width to each image, making the mobile one look much bigger and the desktop one very small.

I played around adding a width attribute to the img element, but it doesn't work either. My options here would be:

- Leaving the image as is.
- Picking only one of the two form factors and cut the other. I think the mobile one would be representative enough.

Let me know what you think.

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 4 above)
Comment (demianrenzulli) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:45]: I've added two stats, please let me know if they look good.

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 5 above)
Comment (demianrenzulli) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:79]: In the original image, mobile occupies 1/3 of the image and desktop 2/3. When breaking into two images, the column component assigns 50% to each, which makes the mobile look much bigger and the desktop much smaller.

I managed to make both look better by adding width="141" to the img element of meli-pdp-mobile.png, but it looks like a hack (perhaps you know a better way of doing this).

In any case, I still like more how it looked like in the original one.

Let me know if you have any ideas for this.

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 11 above)
Comment (demianrenzulli) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:205]: These are two independent optimizations, and shouldn't be added as they were applied to the same asset:

- The 2% reduction is for the tracking module (this impacted the hole site as many sections use it).
- The 16% reduction was for the main bundle used. by PDP pages.

Please, let me know if you think there's a more clear way to explain this.

**Comment by demianrenzulli** [comment]
Hi @kaycebasques! 

Thanks for taking the time of reviewing this PR. I accepted most suggestions and made changes based on them.

Here is a summary of the pending things:

- Stats: I've added two stats via [the stats component](https://web.dev/handbook/web-dev-components/#stats) to highlight the impact in the Lab metric (Max Estimated FID) and in the field (FID). Let me know if these look good to you.

- Mobile/Desktop PDP images: Separated "meli-pdp.png" into two images and organized them with [the column](https://web.dev/handbook/web-dev-components/#columns) component. The end result doesn't look very good in my opinion. I left a comment below yours with more details.

- First Iteration section: Left another comment around the optimizations (the 2% and 16% reductions mentioned at "First Iteration"). Maybe you can help me explain better this part in case it's not clear.

- I used Rewrap to make every line at most 100 chars long. Hope this works.

Regarding the thumbnail, I would have to ask this to the partner. I'm leaving this for the end of the process, in case we have to ask them for anything else (so we can make all requests together).

I hope I haven't missed anything.

Thanks!

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 5 above)
Comment (kaycebasques) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:79]: The columns you've provided look fine. I mainly wanted to split it up in order to add `class="w-screenshot"` to each

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 11 above)
Comment (kaycebasques) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:205]: OK, makes sense. Your current phrasing works.

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 21]
```diff
@@ -0,0 +1,308 @@
...
+layout: post
+title: How Mercado Libre optimized for Web Vitals (TBT/FID)
+subhead: Optimizing interactivity of Mercado Libre's product details pages.
+authors:
+  - demianrenzulli
+  - aranhacarlos
+  - joanbaca
+date: 2020-09-01
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:9]:
```suggestion
date: 2020-09-14
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 4 above)
Comment (kaycebasques) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:45]: LGTM

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 22]
```diff
@@ -0,0 +1,308 @@
...
+<div class="w-columns">
+<figure class="w-figure">
+  <img src="meli-pdp-mobile.png" 
+       alt="Mobile version of a Mercado Libre product detail page." width="141">
+   <figcaption class="w-figcaption">
+      Mobile version of a Mercado Libre product detail page.
+  </figcaption>     
+</figure>
+<figure class="w-figure">
+  <img src="meli-pdp-desktop.png"
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:81]:
```suggestion
  <img src="meli-pdp-desktop.png" class="w-screenshot"
```

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 23]
```diff
@@ -0,0 +1,308 @@
...
+</picture>
+
+While Mercado Libre's site was performing well in most sections, they found in the [Chrome User
+Experience Report](https://developers.google.com/web/tools/chrome-user-experience-report) that
+product detail pages had a poor FID. Based on that information, they decided to focus their efforts
+on improving the interactivity for product pages in the site.
+
+<div class="w-columns">
+<figure class="w-figure">
+  <img src="meli-pdp-mobile.png"
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:74]:
```suggestion
  <img src="meli-pdp-mobile.png" class="w-screenshot"
```

**Review by kaycebasques — CHANGES_REQUESTED** [review]
I think we're almost good-to-go, we just need to sort out the thumbnail

**Comment by kaycebasques** [comment]
@demianrenzulli for the thumbnail I was thinking something like this:

image

**Comment by kaycebasques** [comment]
@demianrenzulli I think this one does have the correct dimensions however (so you can use this one):

thumbnail

**Comment by demianrenzulli** [comment]
Thanks a lot for sharing the thumbnail, and my apologies I didn't have time to create it today!

I'll upload it and let you know.

**Comment by demianrenzulli** [comment]
Hi again @kaycebasques! I just uploaded the thumbnail picture. Please, let me know if it looks good know.

I'm seeing that the mobile and desktop screenshots are misaligned: https://deploy-preview-3814--web-dev-staging.netlify.app/how-mercadolibre-optimized-web-vitals/.

It was difficult to achieve the vertical alignment with the column component. By default it gives 1/2 of the space to each image, when, in reality it should be 1/3 to mobile and 2/3 to desktop (according to the original image). I ended up using a `width="141"` attribute in the `` element of the mobile one, but it doesn't look good in the preview.

Please, let me know if there's any better way to fix this.

**Comment by kaycebasques** [comment]
We can go back to the product detail page screenshots being one image if you're having a lot of trouble with it and it doesn't look right to you.

I cropped out the black border and white border from the images:

pdp

**Comment by kaycebasques** [comment]
Oh, my version looks a bit blurry… well, you get the idea

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 24]
```diff
@@ -0,0 +1,309 @@
+---
+layout: post
+title: How Mercado Libre optimized for Web Vitals (TBT/FID)
+subhead: Optimizing interactivity of Mercado Libre's product details pages.
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:4]:
```suggestion
subhead: Optimizing interactivity of product details pages for a 90% reduction in First Input Delay and a 9% improvement in Chrome User Experience Report.
```
WDYT about mentioning the metric improvement here in the subtitle as well?

**Review by kaycebasques — APPROVED** [review]
I think we're good. I'll let you swap out the PDP image if you want, Demian. Also check out my last suggestion on the subheading.

**Comment by demianrenzulli** [comment]
Hi @kaycebasques, thanks for working on the image once again. If you look how it looks like in your version, maybe you can commit that one?

The issue I was referring to is the following one. If you see the text below the images it looks misaligned:

meli

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md (hunk 24 above)
Comment (demianrenzulli) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:4]: It sounds like a good idea, but it should be: "for a 90% reduction in Reduction in Max Potential FID in Lighthouse and a 9% improvement in FID in Chrome User Experience Report."

I don't think I can apply a change to your suggestion from the UI and commit. Let me know if I should commit it on my end.

Comment (kaycebasques) [reply src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:4]: Fixed!

**Comment by kaycebasques** [comment]
@demianrenzulli I can't commit to your forked branch so you should commit the image

**Comment by kaycebasques** [comment]
We can publish tomorrow or Monday (which supposedly gets a little more initial traffic but I don't think it matters in the long run), LMK what you prefer

**Comment by demianrenzulli** [comment]
Sounds good @kaycebasques! What would be the final suggestion here? Using the original image, without the columns?

If so, I can commit right now. Just let me know.

**Comment by demianrenzulli** [comment]
Or maybe creating a single image, similar to the original, but removing the borders.

**Comment by demianrenzulli** [comment]
Regarding publication: Monday sounds good

**Comment by kaycebasques** [comment]
Use the image I provided (or recreate the image to get rid of the black border around the mobile screenshot) and make sure to add `class="w-screenshot"` 

Sorry for this neverending nitpick haha but we may as well get it right since we've sunk so much time into it already

**Comment by demianrenzulli** [comment]
Thanks @kaycebasques! If we are going to publish on Monday, I'll work on this tomorrow morning, so I make sure it looks good.

File: src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md [hunk 25]
```diff
@@ -0,0 +1,300 @@
+---
+layout: post
+title: How Mercado Libre optimized for Web Vitals (TBT/FID)
+subhead: Optimizing interactivity of product details pages for a 90% reduction in Max Potential FID in Lighthouse and a 9% improvement in FID in Chrome User Experience Report.
+authors:
+  - demianrenzulli
+  - aranhacarlos
+  - joanbaca
+date: 2020-09-15
```
Comment (kaycebasques) [inline src/site/content/en/blog/how-mercadolibre-optimized-web-vitals/index.md:9]:
```suggestion
date: 2020-09-21
```

**Pull request merged**